rust/psibase/src/services/auth_dyn.rs

```diff
@@ -0,0 +1,108 @@
+pub mod interfaces {
+    use fracpack::{Pack, ToSchema, Unpack};
+    use serde::{Deserialize, Serialize};
+
+    use crate::AccountNumber;
+
+    #[derive(Debug, Clone, Serialize, Deserialize, Pack, Unpack, ToSchema)]
+    #[fracpack(fracpack_mod = "fracpack")]
+    pub struct SingleAuth {
+        pub authorizer: AccountNumber,
+    }
+
+    #[derive(Debug, Clone, Serialize, Deserialize, Pack, Unpack, ToSchema)]
+    #[fracpack(fracpack_mod = "fracpack")]
+    pub struct WeightedAuthorizer {
+        pub account: AccountNumber,
+        pub weight: u8,
+    }
+
+    #[derive(Debug, Clone, Serialize, Deserialize, Pack, Unpack, ToSchema)]
+    #[fracpack(fracpack_mod = "fracpack")]
+    pub struct MultiAuth {
+        pub threshold: u8,
+        pub authorizers: Vec<WeightedAuthorizer>,
+    }
+
+    #[derive(Debug, Clone, Serialize, Deserialize, Pack, Unpack, ToSchema)]
+    #[fracpack(fracpack_mod = "fracpack")]
+    pub enum DynamicAuthPolicy {
+        Single(SingleAuth),
+        Multi(MultiAuth),
+    }
+}
+
+#[crate::service(
+    name = "auth-dyn-it",
+    actions = "int_actions",
+    wrapper = "int_wrapper",
+    structs = "int_structs",
+    dispatch = false,
+    pub_constant = false,
+    psibase_mod = "crate"
+)]
+#[allow(non_snake_case, unused_variables)]
+pub mod Int {
```

Comment:
I don't think `Int` is a good name here, maybe just the full service name? `AuthDynIt`?
Although that too could probably be improved.

`AuthDynIntf`?